inspirehep/modules/workflows/tasks/upload.py

```diff
@@ -107,8 +107,14 @@ def store_record_inspirehep_api(obj, eng, is_update, is_authors):
 def send_record_to_hep(obj, pid_type, control_number=None):
     try:
         if control_number:
+            head_version_id = obj.extra_data.get('head_version_id')
```

Comment:
This is set only in Article WF and only if we are doing Merge if I'm correct. What with Article WF and core-selection?

Reply:
edit_article is also missing this key

Reply:
Are you sure, all the updates should have the `head_version_id`

Reply:
we alway `merge_articles` hence it always have `head_version_id` I'm not sure if I'm missing something here

Reply:
For article workflow you are right but for other workflows: 
On edit_article we are not running merge_articles https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/workflows/workflows/edit_article.py
On Core-selection neither https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/workflows/workflows/core_selection.py We might have `head_version_id` here but it's a copy from workflow which creates core-selection and it might be outdated)
on Authors WF we are also not running merge https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/workflows/workflows/author.py (but here I'm not sure if we use this WF for authors updates?)